### ISSUE TITLE
Type hinting a return type as nullable renders `None`

### ISSUE THREAD

**Issue by ViRuSTriNiTy** [pylint-dev/pylint#8603]
### Bug description

Hi there,

the following code

```
class Foo:
    def bar() -> int | None:
        pass
```

produces

![image](https://user-images.githubusercontent.com/12170676/233833424-47808c3f-b4d1-4b79-bb01-827072f70796.png)

meaning `None` is not considered as part of the type hint. My current workaround is to revert back to `Optional[T]` like

```
class Foo:
    def bar() -> Optional[int]:
        pass
```

which correctly produces

![image](https://user-images.githubusercontent.com/12170676/233833467-f52b05a0-8a86-4210-9a33-be9b8e1ec234.png)

I think the handling for the *new* syntax for nullable is missing, which would explain why `Optional[T]` is working but `T | None` is not.

### Configuration

```ini
`pyreverse --version`


pyreverse is included in pylint:
pylint 2.17.2
astroid 2.15.3
Python 3.11.2 (tags/v3.11.2:878ead1, Feb  7 2023, 16:38:35) [MSC v.1934 64 bit (AMD64)]
```
```


### Command used

```shell
`pyreverse -o png -m y src/mt5_proxies`
```

### Pylint output

```shell
See bug description
```

### Expected behavior

Type hint `T | None` is correctly handled like `Option[T]`

### Pylint version

```shell
Python 3.11.2
```

**Comment by ViRuSTriNiTy** [comment]
It seems that there is a general issue with the `X | Y` syntax as using `int | str` produces the same result.

**Comment by ViRuSTriNiTy** [comment]
Hmm, seems like the output type is involved here too. Using the built-in mermaidjs printer produces the diagram correctly:

![image](https://user-images.githubusercontent.com/12170676/233834268-5ff9f596-b2a1-4448-985b-3623ec71c836.png)

**Comment by Pierre-Sassoulas** [comment]
Thank you for the analysis @ViRuSTriNiTy, we probably need to change the rendering to handle the ``|`` operator for renderer that do not support it natively. The code to change would be [here for plantuml](https://github.com/pylint-dev/pylint/blob/main/pylint/pyreverse/plantuml_printer.py) and [here for dot](https://github.com/pylint-dev/pylint/blob/main/pylint Do you want to try ? /pyreverse/dot_printer.py), so probably replace the generic ``pylint.pyreverse.utils.get_annotation_label`` to something specific to each printer.

**Comment by ViRuSTriNiTy** [comment]
@Pierre-Sassoulas Thanks for pointing me into the right direction.

I think I found the line that produces the wrong output, already added some commits in my fork. It is caused by missing escape handling. The docs of the DOT language mention this requirement in section [Record-based Nodes](https://graphviz.org/doc/info/shapes.html#record) with

> Braces, vertical bars and angle brackets must be escaped with a backslash character if you wish them to appear as a literal character.  

I will provide a PR to fix this.
